connectors/sources/outlook.py

```diff
@@ -0,0 +1,1003 @@
+        try:
+            with NamedTemporaryFile(mode="w", delete=False, suffix=".cer") as file:
+                file.write(global_cert)
+
+                ssl_certificate_file = {
+                    global_dns_name: file.name,
+                }[urlparse(url).hostname]
+
+                super().cert_verify(
+                    conn=conn, url=url, verify=ssl_certificate_file, cert=cert
+                )
+        except Exception as exception:
+            raise SSLFailed(
+                f"Something went wrong while verifying SSL certificate. Error: {exception}"
+            ) from exception
+
+
+class ExchangeUsers:
+    """Fetch users from Exchange Active Directory"""
+
+    def __init__(
+        self, ad_server, domain, exchange_server, user, password, ssl_enabled, ssl_ca
+    ):
+        self.ad_server = Server(host=ad_server)
+        self.domain = domain
+        self.exchange_server = exchange_server
+        self.user = user
+        self.password = password
+        self.ssl_enabled = ssl_enabled
+        self.ssl_ca = ssl_ca
+
+    @cached_property
+    def _create_connection(self):
+        return Connection(
+            server=self.ad_server,
+            user=self.user,
+            password=self.password,
+            client_strategy=SAFE_SYNC,
+            auto_bind=True,  # pyright: ignore
+        )
+
+    async def close(self):
+        pass
+
+    def _fetch_normal_users(self, search_query):
+        try:
+            has_value_for_normal_users, _, response, _ = self._create_connection.search(
+                search_query,
+                SEARCH_FILTER_FOR_NORMAL_USERS,
+                attributes=["mail"],
+            )
+
+            if not has_value_for_normal_users:
+                raise UsersFetchFailed(
+                    "Error while fetching users from Exchange Active Directory."
+                )
+
+            for user in response:
+                yield user
+
+        except Exception as e:
+            raise UsersFetchFailed(
+                f"Something went wrong while fetching users. Error: {e}"
+            ) from e
+
+    def _fetch_admin_users(self, search_query):
+        try:
+            (
+                has_value_for_admin_users,
+                _,
+                response_for_admin,
+                _,
+            ) = self._create_connection.search(
+                search_query,
+                SEARCH_FILTER_FOR_ADMIN,
+                attributes=["mail"],
+            )
+
+            if not has_value_for_admin_users:
+                raise UsersFetchFailed(
+                    "Error while fetching users from Exchange Active Directory."
+                )
+
+            for user in response_for_admin:
+                yield user
+        except Exception as e:
+            raise UsersFetchFailed(
+                f"Something went wrong while fetching users. Error: {e}"
+            ) from e
+
+    async def get_users(self):
+        warnings.filterwarnings("default")
```

Comment:
This line has no effect, `warnings.filterwarnings("default")` resets warning filtering to default python settings. If we want to filter some specific warnings refer to my comment from yesterday. If we can leave it as it is without warning filtering this line is redundant and can be deleted.

Reply:
Yes, i think let's not ignore any kind of warnings.